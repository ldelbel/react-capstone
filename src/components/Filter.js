import React from 'react';
import {
  COUNTRIES, SECTORS, EXCHANGES, PRICES,
} from '../constants/index';

const Filter = () => (
  <div style={{
    height: 120, background: '#343a40', borderTopLeftRadius: 20, borderTopRightRadius: 20, padding: 15, paddingRight: 30, paddingLeft: 30,
  }}
  >
    <div><span style={{ fontFamily: '"Alfa Slab One", cursive', color: 'var(--white)', fontSize: 30 }}>Filters&nbsp;</span></div>
    <div className="d-flex flex-row justify-content-center align-items-center" style={{ height: 50 }}>
      <div style={{ marginRight: 20 }}>
        <span style={{
          fontFamily: 'Alata, sans-serif', color: 'var(--light)', fontSize: 20, marginRight: 10,
        }}
        >
          Price Range&nbsp;
        </span>
        <select style={{ borderRadius: 8 }}>
          {PRICES.map(price => (
            <option key={price} value={price} selected>{price}</option>
          ))}

        </select>
      </div>
      <div style={{ marginRight: 20 }}>
        <span style={{
          fontFamily: 'Alata, sans-serif', color: 'var(--light)', fontSize: 20, marginRight: 10,
        }}
        >
          Sector&nbsp;
        </span>
        <select style={{ borderRadius: 8 }}>
          {SECTORS.map(sector => (
            <option key={sector} value={sector} selected>{sector}</option>
          ))}
        </select>
      </div>
      <div style={{ marginRight: 20 }}>
        <span style={{
          fontFamily: 'Alata, sans-serif', color: 'var(--light)', fontSize: 20, marginRight: 10,
        }}
        >
          Country&nbsp;
        </span>
        <select style={{ borderRadius: 8 }}>
          {COUNTRIES.map(country => (
            <option key={country} value={country} selected>{country}</option>
          ))}
        </select>
      </div>
      <div style={{ marginRight: 20 }}>
        <span style={{
          fontFamily: 'Alata, sans-serif', color: 'var(--light)', fontSize: 20, marginRight: 10,
        }}
        >
          Exchange&nbsp;
        </span>
        <select style={{ borderRadius: 8 }}>
          {EXCHANGES.map(exchange => (
            <option key={exchange} value={exchange} selected>{exchange}</option>
          ))}
        </select>
      </div>
      <button
        className="btn btn-primary"
        type="button"
        style={{
          fontFamily: 'Alata, sans-serif', fontSize: 20, color: 'var(--white)', background: 'var(--yellow)', borderStyle: 'none', paddingTop: 3, paddingBottom: 6, fontWeight: 'bold', borderRadius: 15,
        }}
      >
        Apply
      </button>
    </div>
  </div>
);

export default Filter;
