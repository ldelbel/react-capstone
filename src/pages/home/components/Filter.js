import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  COUNTRIES, SECTORS, EXCHANGES, PRICES,
} from '../../../constants/index';

const Filter = props => {
  const [filter, setFilter] = useState({
    price: null,
    sector: null,
    country: null,
    exchange: null,
  });

  const handleChange = target => {
    const { value } = target.options[target.selectedIndex];
    switch (target.name) {
      case 'price':
        setFilter({ ...filter, price: value === 'null' ? null : value });
        return filter;
      case 'sector':
        setFilter({ ...filter, sector: value === 'null' ? null : value });
        return filter;
      case 'country':
        setFilter({ ...filter, country: value === 'null' ? null : value });
        return filter;
      case 'exchange':
        setFilter({ ...filter, exchange: value === 'null' ? null : value });
        return filter;
      default:
        return filter;
    }
  };

  const handleSubmit = () => {
    props.handleUpdateFilter(filter);
  };

  return (
    <div style={{
      height: 120, background: '#343a40', borderTopLeftRadius: 20, borderTopRightRadius: 20, padding: 15, paddingRight: 30, paddingLeft: 30,
    }}
    >
      <div><span style={{ fontFamily: '"Alfa Slab One", cursive', color: 'var(--white)', fontSize: 30 }}>Filters&nbsp;</span></div>
      <div className="d-flex flex-row justify-content-center align-items-center" style={{ height: 50 }}>
        <div style={{ marginRight: 20 }}>
          <span style={{
            fontFamily: 'Alata, sans-serif', color: 'var(--light)', fontSize: 20, marginRight: 10,
          }}
          >
            Price Range&nbsp;
          </span>
          <select
            style={{ borderRadius: 8 }}
            defaultValue="null"
            onChange={e => handleChange(e.target)}
            name="price"
          >
            <option key="uniq" value="null">
              Select
            </option>
            {PRICES.map(price => (
              <option
                key={price}
                value={price}
              >
                {`< ${price}`}
              </option>
            ))}

          </select>
        </div>
        <div style={{ marginRight: 20 }}>
          <span style={{
            fontFamily: 'Alata, sans-serif', color: 'var(--light)', fontSize: 20, marginRight: 10,
          }}
          >
            Sector&nbsp;
          </span>
          <select
            style={{ borderRadius: 8 }}
            onChange={e => handleChange(e.target)}
            name="sector"
            defaultValue="null"
          >
            <option key="uniq" value="null">
              Select
            </option>
            {SECTORS.map(sector => (
              <option key={sector} value={sector}>{sector}</option>
            ))}
          </select>
        </div>
        <div style={{ marginRight: 20 }}>
          <span style={{
            fontFamily: 'Alata, sans-serif', color: 'var(--light)', fontSize: 20, marginRight: 10,
          }}
          >
            Country&nbsp;
          </span>
          <select
            style={{ borderRadius: 8 }}
            onChange={e => handleChange(e.target)}
            name="country"
            defaultValue="null"
          >
            <option key="uniq" value="null">
              Select
            </option>
            {COUNTRIES.map(country => (
              <option key={country} value={country}>{country}</option>
            ))}
          </select>
        </div>
        <div style={{ marginRight: 20 }}>
          <span style={{
            fontFamily: 'Alata, sans-serif', color: 'var(--light)', fontSize: 20, marginRight: 10,
          }}
          >
            Exchange&nbsp;
          </span>
          <select
            style={{ borderRadius: 8 }}
            onChange={e => handleChange(e.target)}
            name="exchange"
            defaultValue="null"
          >
            <option key="uniq" value="null">
              Select
            </option>
            {EXCHANGES.map(exchange => (
              <option key={exchange} value={exchange}>{exchange}</option>
            ))}
          </select>
        </div>
        <button
          className="btn"
          type="button"
          onClick={handleSubmit}
          style={{
            fontFamily: 'Alata, sans-serif', fontSize: 20, color: 'var(--white)', background: 'var(--yellow)', borderStyle: 'none', paddingTop: 3, paddingBottom: 6, fontWeight: 'bold', borderRadius: 15,
          }}
        >
          Apply
        </button>
      </div>
    </div>
  );
};

Filter.propTypes = {
  handleUpdateFilter: PropTypes.func.isRequired,
};

export default Filter;
