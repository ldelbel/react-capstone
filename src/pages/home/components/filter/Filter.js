import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  COUNTRIES, SECTORS, EXCHANGES, PRICES,
} from '../../../../constants/index';
import styles from './Filter.module.scss';

const Filter = props => {
  const [filter, setFilter] = useState({
    price: null,
    sector: null,
    country: null,
    exchange: null,
  });

  const handleChange = target => {
    const { value } = target.options[target.selectedIndex];
    switch (target.name) {
      case 'price':
        setFilter({ ...filter, price: value === 'null' ? null : value });
        return filter;
      case 'sector':
        setFilter({ ...filter, sector: value === 'null' ? null : value });
        return filter;
      case 'country':
        setFilter({ ...filter, country: value === 'null' ? null : value });
        return filter;
      case 'exchange':
        setFilter({ ...filter, exchange: value === 'null' ? null : value });
        return filter;
      default:
        return filter;
    }
  };

  const handleSubmit = () => {
    props.handleUpdateFilter(filter);
  };

  return (
    <div className={styles.filter}>
      <div className={styles.filter__title}><span>Filters&nbsp;</span></div>
      <div className={`d-flex flex-row justify-content-center align-items-center ${styles.filter__body}`}>
        <div className={styles.filter__body__div}>
          <span>
            Price Range&nbsp;
          </span>
          <select
            defaultValue="null"
            onChange={e => handleChange(e.target)}
            name="price"
          >
            <option key="uniq" value="null">
              Select
            </option>
            {PRICES.map(price => (
              <option
                key={price}
                value={price}
              >
                {`< ${price}`}
              </option>
            ))}
          </select>
        </div>
        <div className={styles.filter__body__div}>
          <span>
            Sector&nbsp;
          </span>
          <select
            onChange={e => handleChange(e.target)}
            name="sector"
            defaultValue="null"
          >
            <option key="uniq" value="null">
              Select
            </option>
            {SECTORS.map(sector => (
              <option key={sector} value={sector}>{sector}</option>
            ))}
          </select>
        </div>
        <div className={styles.filter__body__div}>
          <span>
            Country&nbsp;
          </span>
          <select
            onChange={e => handleChange(e.target)}
            name="country"
            defaultValue="null"
          >
            <option key="uniq" value="null">
              Select
            </option>
            {COUNTRIES.map(country => (
              <option key={country} value={country}>{country}</option>
            ))}
          </select>
        </div>
        <div className={styles.filter__body__div}>
          <span>
            Exchange&nbsp;
          </span>
          <select
            onChange={e => handleChange(e.target)}
            name="exchange"
            defaultValue="null"
          >
            <option key="uniq" value="null">
              Select
            </option>
            {EXCHANGES.map(exchange => (
              <option key={exchange} value={exchange}>{exchange}</option>
            ))}
          </select>
        </div>
        <button
          className={`btn ${styles.filter__body__btn}`}
          type="button"
          onClick={handleSubmit}
        >
          Apply
        </button>
      </div>
    </div>
  );
};

Filter.propTypes = {
  handleUpdateFilter: PropTypes.func.isRequired,
};

export default Filter;
