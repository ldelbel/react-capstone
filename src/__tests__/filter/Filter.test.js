import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Filter from '../../components/Filter';

it('renders correctly', () => {
  const navbar = renderer.create(
    <BrowserRouter>
      <Filter />
    </BrowserRouter>,
  ).toJSON();
  expect(navbar).toMatchSnapshot();
});