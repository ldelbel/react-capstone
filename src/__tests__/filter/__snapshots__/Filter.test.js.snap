// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter DOM renders correctly 1`] = `
<div
  className="filter"
>
  <div
    className="filter__title"
  >
    <span>
      Filters 
    </span>
  </div>
  <div
    className="d-flex flex-row justify-content-center align-items-center filter__body"
  >
    <div
      className="filter__body__div"
    >
      <span>
        Price Range 
      </span>
      <select
        defaultValue="null"
        name="price"
        onChange={[Function]}
      >
        <option
          value="null"
        >
          Select
        </option>
        <option
          value={100}
        >
          &lt; 100
        </option>
        <option
          value={200}
        >
          &lt; 200
        </option>
        <option
          value={300}
        >
          &lt; 300
        </option>
        <option
          value={500}
        >
          &lt; 500
        </option>
        <option
          value={1000}
        >
          &lt; 1000
        </option>
        <option
          value={2000}
        >
          &lt; 2000
        </option>
      </select>
    </div>
    <div
      className="filter__body__div"
    >
      <span>
        Sector 
      </span>
      <select
        defaultValue="null"
        name="sector"
        onChange={[Function]}
      >
        <option
          value="null"
        >
          Select
        </option>
        <option
          value="Aerospace & Defense"
        >
          Aerospace & Defense
        </option>
        <option
          value="Airlines"
        >
          Airlines
        </option>
        <option
          value="Auto Components"
        >
          Auto Components
        </option>
        <option
          value="Automobiles"
        >
          Automobiles
        </option>
        <option
          value="Banking"
        >
          Banking
        </option>
        <option
          value="Basic Materials"
        >
          Basic Materials
        </option>
        <option
          value="Beverages"
        >
          Beverages
        </option>
        <option
          value="Biotechnology"
        >
          Biotechnology
        </option>
        <option
          value="Building"
        >
          Building
        </option>
        <option
          value="Chemicals"
        >
          Chemicals
        </option>
        <option
          value="Commercial Services & Supplies"
        >
          Commercial Services & Supplies
        </option>
        <option
          value="Communication Services"
        >
          Communication Services
        </option>
        <option
          value="Communications"
        >
          Communications
        </option>
        <option
          value="Construction"
        >
          Construction
        </option>
        <option
          value="Consumer Cyclical"
        >
          Consumer Cyclical
        </option>
        <option
          value="Consumer Defensive"
        >
          Consumer Defensive
        </option>
        <option
          value="Consumer products"
        >
          Consumer products
        </option>
        <option
          value="Distributors"
        >
          Distributors
        </option>
        <option
          value="Diversified Consumer Services"
        >
          Diversified Consumer Services
        </option>
        <option
          value="Electrical Equipment"
        >
          Electrical Equipment
        </option>
        <option
          value="Energy"
        >
          Energy
        </option>
        <option
          value="Financial"
        >
          Financial
        </option>
        <option
          value="Financial Services"
        >
          Financial Services
        </option>
        <option
          value="Food Products"
        >
          Food Products
        </option>
        <option
          value="Health Care"
        >
          Health Care
        </option>
        <option
          value="Healthcare"
        >
          Healthcare
        </option>
        <option
          value="Hotels, Restaurants & Leisure"
        >
          Hotels, Restaurants & Leisure
        </option>
        <option
          value="Industrial Conglomerates"
        >
          Industrial Conglomerates
        </option>
        <option
          value="Industrial Goods"
        >
          Industrial Goods
        </option>
        <option
          value="Industrials"
        >
          Industrials
        </option>
        <option
          value="Insurance"
        >
          Insurance
        </option>
        <option
          value="Leisure Products"
        >
          Leisure Products
        </option>
        <option
          value="Life Sciences Tools & Services"
        >
          Life Sciences Tools & Services
        </option>
        <option
          value="Logistics & Transportation"
        >
          Logistics & Transportation
        </option>
        <option
          value="Machinery"
        >
          Machinery
        </option>
        <option
          value="Marine"
        >
          Marine
        </option>
        <option
          value="Media"
        >
          Media
        </option>
        <option
          value="Metals & Mining"
        >
          Metals & Mining
        </option>
        <option
          value="Packaging"
        >
          Packaging
        </option>
        <option
          value="Paper & Forest"
        >
          Paper & Forest
        </option>
        <option
          value="Pharmaceuticals"
        >
          Pharmaceuticals
        </option>
        <option
          value="Professional Services"
        >
          Professional Services
        </option>
        <option
          value="Real Estate"
        >
          Real Estate
        </option>
        <option
          value="Retail"
        >
          Retail
        </option>
        <option
          value="Road & Rail"
        >
          Road & Rail
        </option>
        <option
          value="Semiconductors"
        >
          Semiconductors
        </option>
        <option
          value="Technology"
        >
          Technology
        </option>
        <option
          value="Telecommunication"
        >
          Telecommunication
        </option>
        <option
          value="Textiles, Apparel & Luxury Goods"
        >
          Textiles, Apparel & Luxury Goods
        </option>
        <option
          value="Tobacco"
        >
          Tobacco
        </option>
        <option
          value="Trading Companies & Distributors"
        >
          Trading Companies & Distributors
        </option>
        <option
          value="Transportation Infrastructure"
        >
          Transportation Infrastructure
        </option>
        <option
          value="Utilities"
        >
          Utilities
        </option>
      </select>
    </div>
    <div
      className="filter__body__div"
    >
      <span>
        Country 
      </span>
      <select
        defaultValue="null"
        name="country"
        onChange={[Function]}
      >
        <option
          value="null"
        >
          Select
        </option>
        <option
          value="US"
        >
          US
        </option>
        <option
          value="MX"
        >
          MX
        </option>
        <option
          value="BR"
        >
          BR
        </option>
        <option
          value="GB"
        >
          GB
        </option>
        <option
          value="CN"
        >
          CN
        </option>
        <option
          value="FI"
        >
          FI
        </option>
        <option
          value="NL"
        >
          NL
        </option>
        <option
          value="SG"
        >
          SG
        </option>
        <option
          value="IN"
        >
          IN
        </option>
        <option
          value="CA"
        >
          CA
        </option>
        <option
          value="ZA"
        >
          ZA
        </option>
        <option
          value="BM"
        >
          BM
        </option>
        <option
          value="DE"
        >
          DE
        </option>
        <option
          value="IE"
        >
          IE
        </option>
        <option
          value="CH"
        >
          CH
        </option>
        <option
          value="LU"
        >
          LU
        </option>
        <option
          value="IL"
        >
          IL
        </option>
        <option
          value="HK"
        >
          HK
        </option>
        <option
          value="TW"
        >
          TW
        </option>
        <option
          value="ES"
        >
          ES
        </option>
        <option
          value="FR"
        >
          FR
        </option>
        <option
          value="MC"
        >
          MC
        </option>
        <option
          value="GR"
        >
          GR
        </option>
        <option
          value="KR"
        >
          KR
        </option>
        <option
          value="PR"
        >
          PR
        </option>
        <option
          value="SE"
        >
          SE
        </option>
        <option
          value="AU"
        >
          AU
        </option>
        <option
          value="RU"
        >
          RU
        </option>
        <option
          value="KY"
        >
          KY
        </option>
        <option
          value="CL"
        >
          CL
        </option>
        <option
          value="CO"
        >
          CO
        </option>
        <option
          value="DK"
        >
          DK
        </option>
        <option
          value="BE"
        >
          BE
        </option>
        <option
          value="JP"
        >
          JP
        </option>
        <option
          value="AR"
        >
          AR
        </option>
        <option
          value="UY"
        >
          UY
        </option>
        <option
          value="CZ"
        >
          CZ
        </option>
        <option
          value="VI"
        >
          VI
        </option>
        <option
          value="JE"
        >
          JE
        </option>
        <option
          value="PA"
        >
          PA
        </option>
        <option
          value="TH"
        >
          TH
        </option>
        <option
          value="ID"
        >
          ID
        </option>
        <option
          value="IT"
        >
          IT
        </option>
        <option
          value="TR"
        >
          TR
        </option>
        <option
          value="PH"
        >
          PH
        </option>
        <option
          value="CY"
        >
          CY
        </option>
        <option
          value="GG"
        >
          GG
        </option>
        <option
          value="MO"
        >
          MO
        </option>
        <option
          value="NZ"
        >
          NZ
        </option>
        <option
          value="NO"
        >
          NO
        </option>
        <option
          value="PE"
        >
          PE
        </option>
        <option
          value="CR"
        >
          CR
        </option>
        <option
          value="BS"
        >
          BS
        </option>
        <option
          value="PT"
        >
          PT
        </option>
        <option
          value="SN"
        >
          SN
        </option>
        <option
          value="CI"
        >
          CI
        </option>
        <option
          value="IS"
        >
          IS
        </option>
        <option
          value="MA"
        >
          MA
        </option>
        <option
          value="CW"
        >
          CW
        </option>
        <option
          value="IM"
        >
          IM
        </option>
        <option
          value="GA"
        >
          GA
        </option>
        <option
          value="ZM"
        >
          ZM
        </option>
        <option
          value="CK"
        >
          CK
        </option>
        <option
          value="VG"
        >
          VG
        </option>
        <option
          value="AE"
        >
          AE
        </option>
        <option
          value="JO"
        >
          JO
        </option>
        <option
          value="MT"
        >
          MT
        </option>
        <option
          value="AT"
        >
          AT
        </option>
        <option
          value="PL"
        >
          PL
        </option>
        <option
          value="FO"
        >
          FO
        </option>
        <option
          value="LI"
        >
          LI
        </option>
        <option
          value="GI"
        >
          GI
        </option>
        <option
          value="AZ"
        >
          AZ
        </option>
        <option
          value="FK"
        >
          FK
        </option>
        <option
          value="UA"
        >
          UA
        </option>
        <option
          value="MY"
        >
          MY
        </option>
        <option
          value="MU"
        >
          MU
        </option>
        <option
          value="TC"
        >
          TC
        </option>
        <option
          value="NG"
        >
          NG
        </option>
        <option
          value="MN"
        >
          MN
        </option>
        <option
          value="TG"
        >
          TG
        </option>
        <option
          value="GE"
        >
          GE
        </option>
        <option
          value="VN"
        >
          VN
        </option>
        <option
          value="HU"
        >
          HU
        </option>
        <option
          value="KH"
        >
          KH
        </option>
        <option
          value="SB"
        >
          SB
        </option>
        <option
          value="PG"
        >
          PG
        </option>
        <option
          value="BB"
        >
          BB
        </option>
        <option
          value="BD"
        >
          BD
        </option>
        <option
          value="DO"
        >
          DO
        </option>
        <option
          value="SK"
        >
          SK
        </option>
        <option
          value="AI"
        >
          AI
        </option>
        <option
          value="LT"
        >
          LT
        </option>
        <option
          value="BG"
        >
          BG
        </option>
      </select>
    </div>
    <div
      className="filter__body__div"
    >
      <span>
        Exchange 
      </span>
      <select
        defaultValue="null"
        name="exchange"
        onChange={[Function]}
      >
        <option
          value="null"
        >
          Select
        </option>
        <option
          value="NYSE"
        >
          NYSE
        </option>
        <option
          value="NASDAQ"
        >
          NASDAQ
        </option>
        <option
          value="AMEX"
        >
          AMEX
        </option>
        <option
          value="EURONEX"
        >
          EURONEX
        </option>
        <option
          value="TSX"
        >
          TSX
        </option>
        <option
          value="ETF"
        >
          ETF
        </option>
        <option
          value="MUTUAL FUND"
        >
          MUTUAL FUND
        </option>
      </select>
    </div>
    <button
      className="btn filter__body__btn"
      onClick={[Function]}
      type="button"
    >
      Apply
    </button>
  </div>
</div>
`;
